snekbox/api/resources/eval.py

```diff
@@ -38,8 +49,6 @@ def on_post(self, req: falcon.Request, resp: falcon.Response) -> None:
         Evaluate Python code and return stdout, stderr, and the return code.
 
         A list of arguments for the Python subprocess can be specified as `args`.
-        Otherwise, the default argument "-c" is used to execute the input code.
-        The input code is always passed as the last argument to Python.
```

Comment:
Seems reasonable yeah, I'll look into compatible support for "input"

Reply:
should we have args implicitly include an "-c" default if "input" is non-empty?

Reply:
If `input` exists, it should become the last Python argument. If `input` exists and `args` does not exist, then `args` should default to `["-c"]` (in addition to the `input` becoming the last Python argument).

Reply:
Honestly after attempting an implementation, this seems to just increase complexity and error-prone code (like the fact that our specification will be an 'either-or-but-at-least-one' requirement of `args` and `input`. The initial inclusion of `input` was already largely redundant as it is directly appended to `args`, so `args` can achieve all of the previous usages of `input` but without the ambiguous parsing and dynamic default of "-c" that we do on our end. 

In the future, if we support requests specifying user input via `sys.argv`, the name occupied by `input` will be more ambiguous and confusing. 

In general the deprecation of `input` at this stage seems like a reasonable path. If we truly need to support users beyond python bot, imo the usual case is maintaining a versioned endpoint that retains previous behavior and specifications fully, instead of retaining it as an optional parameter in the same endpoint (Which I've never really seen as a path taken by any major public API).

For the CLI interface, the simplicity in typing the code directly could be made, but for an API, we're retaining a parameter (input) just to be later conditionally appended to an already existing parameter (args) which has a default that conditionally exists based on whether the new parameter is empty or not. With all of this being able to be avoided from the requester supplying a single explicit "-c" in `args`.

Considering the additional keys in the response is already potentially breaking, and our numerous breaking library-side changes (such as the return types and arguments of `python3`), I don't really see a library side breaking concern more than any other change. We should increment a major version, and downstream users can upgrade when they're ready.

Reply:
I've added compatible support for input in the API route in 84e73c1, so it is backwards compatible now. Added some associated unit tests to ensure the old `input` style still works.

Currently the API route requires either `args` or `input`. If `input` exists, `args` will have a default of `-c` if it is otherwise empty.